### DISCUSSION TITLE
Inconsistent WiFi components in multiplayer

### DISCUSSION THREAD

**Discussion by Hapunn** [FakeFishGames/Barotrauma#13976] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

I was originally working on a fixfoam-detonator system using WiFi and RegEx components to detect hull breach quick reports from chat. I was making the initial logic in a circuit box but ran into inconsistency with the WiFi components; they all seemed to not want to listen to certain channels. I thought it was an issue with my RegEx formatting but even simple test phrases like "test1" and so forth didn't work either.
I moved the schematic to a wall outside the circuit box and it still didn't work, so I set up an incredibly basic circuit to narrow down the issue and it seems to me that sometimes WiFi components just decide to not receive any input from their respective radio channel in multiplayer.

Pictured below is the simple circuit in multiplayer, a WiFi component linked to chat in channel 0, with it's output sent to a text display's SET_TEXT. No luck.
![image](https://github.com/FakeFishGames/Barotrauma/assets/170216118/85fce305-a856-48af-8803-3db5e40028d6)

The same circuit worked flawlessly, as expected, in a simple sub editor run.
![image](https://github.com/FakeFishGames/Barotrauma/assets/170216118/588760fa-9d0d-4c54-bdb2-341b7179f247)

### Reproduction steps

1. Start a multiplayer game with about 5 other people
_1.5. Maybe be deep into the campaign with many objects on the sub and monster entities nearby_ 
2. Attempt to create the simple WiFi component to text display circuit above
3. Profit ???

### Bug prevalence

Just once

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by BlackMoons** [comment]
Make sure you have the 'allow link to chat' setting in your server enabled.

**Reply by Hapunn** [reply]
Yes, that was enabled. Forgot to mention that

**Comment by zeroinformatique** [comment]
We had a similar bug in our game. Was fiddling with WIFI but it never worked properly.

We quit the session, restart it, and suddenly the WIFI worked perfectly fine.

**Comment by Regalis11** [comment]
Addresed in https://github.com/FakeFishGames/Barotrauma-development/pull/5653 (PR in our private repo)

**Reply by Hapunn** [reply]
Lovely, thank you!

**Comment by Jasontti** [comment]
Tested a fix for this.
Closing discussion as resolved.
